### PR TITLE
DailyService

### PR DESCRIPTION
**Pull request by Drummss**
### Implementation of a DailyService
There is now a DailyService that will handle all things daily. Along with this, I added full localization for the daily command.

### CODE REVIEW

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 1]
```diff
@@ -1033,30 +1033,29 @@ await e.ErrorEmbedResource("give_error_amount_unparsable")
         [Command("daily")]
         public async Task GetDailyAsync(IContext e)
         {
-            var context = e.GetService<MikiDbContext>();
-
+            MikiDbContext context = e.GetService<MikiDbContext>();
             User u = await DatabaseHelpers.GetUserAsync(context, e.GetAuthor());
 
-            if(u == null)
+            if(u == null) // Check it the user exists.
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: What are you trying to explain with these comments that I cannot deduce from looking at the code?

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 2]
```diff
@@ -1077,39 +1076,51 @@ await builder.ToEmbed()
                 return;
             }
 
-            int streak = 0;
+            /* Streak check. */
+            DailyStreak dailyStreak = await DailyStreak.GetAsync(context, u.Id);
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Preferably we want to have no new raw Psql queries, please give a look at the UnitOfWork and Repository pattern. :-)

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 3]
```diff
@@ -1077,39 +1076,51 @@ await builder.ToEmbed()
...
+            ICacheClient cache = e.GetService<ICacheClient>();
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Please use `var` here

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 4]
```diff
@@ -1077,39 +1076,51 @@ await builder.ToEmbed()
...
             string redisKey = $"user:{e.GetAuthor().Id}:daily";
 
-            var cache = e.GetService<ICacheClient>();
+            if (await cache.ExistsAsync(redisKey)) // Check if the user had a cached streak and migrate to database.
+            {
+                dailyStreak.CurrentStreak = await cache.GetAsync<int>(redisKey);
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Please use `GetAsync(...).ConfiguresAwait(false);`

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 5]
```diff
@@ -1077,39 +1076,51 @@ await builder.ToEmbed()
...
             }
 
-            int streak = 0;
+            /* Streak check. */
+            DailyStreak dailyStreak = await DailyStreak.GetAsync(context, u.Id);
+            ICacheClient cache = e.GetService<ICacheClient>();
             string redisKey = $"user:{e.GetAuthor().Id}:daily";
 
-            var cache = e.GetService<ICacheClient>();
+            if (await cache.ExistsAsync(redisKey)) // Check if the user had a cached streak and migrate to database.
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Please use `ExistsAsync(...).ConfiguresAwait(false);`

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 6]
```diff
@@ -1033,30 +1033,29 @@ await e.ErrorEmbedResource("give_error_amount_unparsable")
...
-            if(u == null)
+            if(u == null) // Check it the user exists.
             {
                 await e.ErrorEmbed(e.GetLocale().GetString("user_error_no_account"))
                     .ToEmbed().QueueAsync(e, e.GetChannel());
                 return;
             }
 
+            /* Base variables */
             int dailyAmount = 100;
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: A good thing to do with these is to migrate them to constant values instead.

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 7]
```diff
@@ -1033,30 +1033,29 @@ await e.ErrorEmbedResource("give_error_amount_unparsable")
         [Command("daily")]
         public async Task GetDailyAsync(IContext e)
         {
-            var context = e.GetService<MikiDbContext>();
-
+            MikiDbContext context = e.GetService<MikiDbContext>();
             User u = await DatabaseHelpers.GetUserAsync(context, e.GetAuthor());
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: This is an old, deprecated way of getting users. The current prefered way is 

```cs
var userService = e.GetService<UserService>();
var user = userService.GetUserAsync(...);
```

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 8]
```diff
@@ -1033,30 +1033,29 @@ await e.ErrorEmbedResource("give_error_amount_unparsable")
...
                     .ToEmbed().QueueAsync(e, e.GetChannel());
                 return;
             }
 
+            /* Base variables */
             int dailyAmount = 100;
             int dailyStreakAmount = 20;
 
-            if(await u.IsDonatorAsync(context))
+            if (await u.IsDonatorAsync(context)) // Check donator status.
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Please use `IsDonatorAsync(...).ConfiguresAwait(false);` And also please look at the donator module for a more up to date approach on checking if a user is a donator.

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 9]
```diff
@@ -1077,39 +1076,51 @@ await builder.ToEmbed()
...
-                    $"**{amount:N0}**", 
+                    $"**{finalAmount:N0}**", 
                     $"`{u.Currency.ToFormattedString()}`"))
                 .SetColor(253, 216, 136);
 
-            if(streak > 0)
+            if(dailyStreak.CurrentStreak > 0)
             {
-                embed.AddInlineField("Streak!", $"You're on a {streak.ToFormattedString()} day daily streak!");
+                embed.AddInlineField("Streak!", $"You're on a {dailyStreak.CurrentStreak.ToFormattedString()} day daily streak!");
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: preferably use `{dailyStreak.CurrentStreak:N0}`.

File: src/Miki/Modules/DeveloperModule.cs [hunk 10]
```diff
@@ -339,5 +340,44 @@ public async Task BanUserAsync(IContext e)
...
+
+        [Command("dailyedit")]
+        [RequiresScope("developer")]
+        public class DailyEditCommand
+        {
+            [Command("resettimer")]
+            [RequiresScope("developer")]
+            public async Task ResetDailyAsync(IContext e)
+            {
+                MikiDbContext context = e.GetService<MikiDbContext>();
```
Comment (velddev) [inline src/Miki/Modules/DeveloperModule.cs]: If it is possible, avoid using raw database handling inside of the business context. I know it's still very much used everywhere, but there's an effort to no longer do this.

**Review by velddev — CHANGES_REQUESTED** [review]

File: src/Miki/Modules/Accounts/AccountsModule.cs (hunk 2 above)
Comment (Drummss) [reply src/Miki/Modules/Accounts/AccountsModule.cs]: I think I'll need a bit of help with this but I have finished all of the other requested changes.

File: src/Miki/Services/StreakService.cs [hunk 11]
```diff
@@ -0,0 +1,55 @@
+﻿namespace Miki.Services
+{
+    using System;
+    using System.Threading.Tasks;
+
```
Comment (velddev) [inline src/Miki/Services/StreakService.cs]: What's with the spacing?

File: src/Miki/Services/StreakService.cs [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    using Bot.Models;
+
+    using Framework;
```
Comment (velddev) [inline src/Miki/Services/StreakService.cs]: Always use the full namespace using :-)

File: src/Miki/Services/StreakService.cs [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+    using Patterns.Repositories;
+
+    public class StreakService : IStreakService
```
Comment (velddev) [inline src/Miki/Services/StreakService.cs]: There's a few issues with your service; It does not seem to elevate internal business logic to the service, but rather seem to work like a repository. I would recommend building a method that would take a user daily streak and increase it if valid. 

e.x.

```cs
public void IncreaseStreak(ulong userId) {
	var user = userService.GetUserById(userId);
	var streak = repository.Get(userId);
	if(IsValid(streak)) 
	{
		streak.value++;
 	}
	else 
	{
		streak = new DailyStreak();
	}
	repository.Edit(streak);

	transactionService.CreateTransaction(new TR { 
		Sender = 0L // Constant for the "miki bank"
		Receiver = (long)userId
		Amount = GetCurrentDailyAmount()
	});
	
	return streak;
}

File: src/Miki/Services/UserService.cs [hunk 14]
```diff
@@ -68,7 +78,9 @@ public interface IUserService : IDisposable
 
         ValueTask UpdateUserAsync(User user);
 
-        ValueTask<bool> UserIsBanned(long userId);
+        ValueTask<bool> UserIsBannedAsync(long userId);
```
Comment (velddev) [inline src/Miki/Services/UserService.cs]: Thanks for the naming refactor here +1

File: src/Miki/Services/UserService.cs [hunk 15]
```diff
@@ -50,16 +49,27 @@ public async ValueTask<bool> UserIsBanned(long userId)
...
+        /// <inheritdoc />
+        public ValueTask SaveAsync()
+            => unitOfWork.CommitAsync();
+
         /// <inheritdoc />
         public void Dispose()
-        {
-            unitOfWork?.Dispose();
-        }
+            => unitOfWork?.Dispose();
```
Comment (velddev) [inline src/Miki/Services/UserService.cs:85]: Unit of work should never be null, avoid using `?` as it should throw an internal exception instead.

File: src/Miki/Services/UserService.cs [hunk 16]
```diff
@@ -50,16 +49,27 @@ public async ValueTask<bool> UserIsBanned(long userId)
         }
 
         /// <inheritdoc />
-        public ValueTask SaveAsync()
+        public async ValueTask<bool> UserIsDonatorAsync(long userId)
         {
-            return unitOfWork.CommitAsync();
+            var donatorRecord = await donatorRepository.GetAsync(userId);
+            if (donatorRecord == null)
```
Comment (velddev) [inline src/Miki/Services/UserService.cs]: replace with `return donatorRecord != null`

File: src/Miki/Modules/DeveloperModule.cs [hunk 17]
```diff
@@ -339,5 +343,48 @@ public async Task BanUserAsync(IContext e)
...
 			}
 		}
-	}
+
+        [Command("dailyedit")]
+        [RequiresScope("developer")]
+        public class DailyEditCommand
+        {
+            [Command("resettimer")]
+            //[RequiresScope("developer")]
```
Comment (velddev) [inline src/Miki/Modules/DeveloperModule.cs]: Don't commit commented code like this, instead make yourself a developer on your own database.

File: src/Miki/Modules/DeveloperModule.cs [hunk 18]
```diff
@@ -339,5 +343,48 @@ public async Task BanUserAsync(IContext e)
 					.BanAsync(context);
 			}
 		}
-	}
+
+        [Command("dailyedit")]
+        [RequiresScope("developer")]
```
Comment (velddev) [inline src/Miki/Modules/DeveloperModule.cs:347]: Sadly this doesn't work yet 😔

Comment (velddev) [reply src/Miki/Modules/DeveloperModule.cs:347]: Please make a feature request in https://github.com/mikibot/miki.framework

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 19]
```diff
@@ -1140,39 +1141,46 @@ await builder.ToEmbed()
...
+            await userService.SaveAsync().ConfigureAwait(false);
 
             var embed = new EmbedBuilder()
                 .SetTitle("💰 Daily")
                 .SetDescription(e.GetLocale().GetString(
                     "daily_received", 
-                    $"**{amount:N0}**", 
-                    $"`{u.Currency.ToFormattedString()}`"))
+                    $"**{finalAmount:N0}**", 
+                    $"`{user.Currency.ToFormattedString()}`"))
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Replace `.ToFormattedString()` with `:N0`. Thanks! ❤️

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 20]
```diff
@@ -1140,39 +1141,46 @@ await builder.ToEmbed()
...
+            } else
             {
-                streak = await cache.GetAsync<int>(redisKey);
-                streak++;
+                dailyStreak.CurrentStreak = 0;
+                dailyStreak.LastStreakTime = DateTime.UtcNow;
             }
 
-            int amount = dailyAmount + (dailyStreakAmount * Math.Min(100, streak));
+            int finalAmount = (dailyAmount * donatorMultiplier) + ((dailyStreakAmount * donatorMultiplier) * Math.Min(100, (int)dailyStreak.CurrentStreak));
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: This seems to be a very long and complex line of code, might be useful to kind of move it to the service layer instead.

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 21]
```diff
@@ -1140,39 +1141,46 @@ await builder.ToEmbed()
...
+                dailyStreak.CurrentStreak = await cache.GetAsync<int>(redisKey).ConfigureAwait(false);
+                await cache.RemoveAsync(redisKey);
+            }
 
-            if(await cache.ExistsAsync(redisKey))
+            if (DateTime.UtcNow <= dailyStreak.LastStreakTime.AddHours(48))
+            {
+                dailyStreak.CurrentStreak++;
+                dailyStreak.LastStreakTime = DateTime.UtcNow;
+            } else
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: `else` should be on a newline

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 22]
```diff
@@ -1140,39 +1141,46 @@ await builder.ToEmbed()
                 return;
             }
 
-            int streak = 0;
-            string redisKey = $"user:{e.GetAuthor().Id}:daily";
+            ICacheClient cache = e.GetService<ICacheClient>();
+            string redisKey = $"user:{user.Id}:daily";
 
-            var cache = e.GetService<ICacheClient>();
+            if (await cache.ExistsAsync(redisKey).ConfigureAwait(false))
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: Avoid using cache operations directly in commands. Rather, let the service handle it

File: src/Miki/MikiBotApp.cs [hunk 23]
```diff
@@ -6,12 +6,17 @@
     using System.Reflection;
     using System.Resources;
     using System.Threading.Tasks;
+
```
Comment (velddev) [inline src/Miki/MikiBotApp.cs]: Please keep the current style consistent with the already defined style

File: src/Miki/Miki.csproj [hunk 24]
```diff
@@ -43,6 +43,7 @@
     <PackageReference Include="NCalc.NetCore" Version="1.0.1" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
     <PackageReference Include="Npgsql" Version="4.1.2" />
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="3.0.1" />
```
Comment (velddev) [inline src/Miki/Miki.csproj]: Why was this required here?

**Review by velddev — CHANGES_REQUESTED** [review]
It seems to be improving a lot, however; we're not there yet. 👍 great progress man!

File: src/Miki/Services/UserService.cs (hunk 15 above)
Comment (Drummss) [reply src/Miki/Services/UserService.cs:85]: That's your code tbh.

File: src/Miki/Services/UserService.cs (hunk 14 above)
Comment (Drummss) [reply src/Miki/Services/UserService.cs]: 👍

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 14
           

Complexity increasing per file
==============================
- test/Miki.Tests/Services/DailyServiceTests.cs  1
- src/Miki/Services/DailyService.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

File: src/Miki/Languages/eng.resx [hunk 25]
```diff
@@ -2109,7 +2109,7 @@ To give someone reputation, do `&gt;rep &lt;mention&gt; ["all"|amount]`</value>
   <data name="error_argument_less_than_zero" xml:space="preserve">
     <value>You can not use an amount less than zero!</value>
   </data>
-  <data name="daily_received" xml:space="preserve">
+  <data name="miki_module_accounts_daily_received" xml:space="preserve">
```
Comment (velddev) [inline src/Miki/Languages/eng.resx]: This is an old method of handling language strings; go for the shorter version instead.

File: src/Miki/Languages/eng.resx [hunk 26]
```diff
@@ -2308,4 +2308,23 @@ To read more about the miki scripting language. Check out https://github.com/mik
   <data name="error_pasta_null" xml:space="preserve">
     <value>Pasta was not found.</value>
   </data>
+  <data name="miki_module_accounts_daily_appreciate_list" xml:space="preserve">
```
Comment (velddev) [inline src/Miki/Languages/eng.resx]: Same for these

Comment (velddev) [reply src/Miki/Languages/eng.resx]: Also; this one is not only used in daily.

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 27]
```diff
@@ -1136,88 +1138,43 @@ private void ThrowOnUserNull(IDiscordGuildUser user)
...
-            await transactionService.CreateTransactionAsync(
-                new TransactionRequest.Builder()
-                    .WithAmount(amount)
-                    .WithReceiver(u.Id)
-                    .WithSender(0L)
-                    .Build());
-
             var embed = new EmbedBuilder()
-                .SetTitle("💰 Daily")
+                .SetTitle(e.GetLocale().GetString("miki_module_accounts_daily_title"))
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: replace to `daily_claimed_title`

File: src/Miki/Modules/Accounts/AccountsModule.cs [hunk 28]
```diff
@@ -1136,88 +1138,43 @@ private void ThrowOnUserNull(IDiscordGuildUser user)
...
                 .SetDescription(e.GetLocale().GetString(
-                    "daily_received", 
-                    $"**{amount:N0}**", 
-                    $"`{(u.Currency):N0}`"))
+                    "miki_module_accounts_daily_received",
```
Comment (velddev) [inline src/Miki/Modules/Accounts/AccountsModule.cs]: replace to `daily_claimed`

File: src/Miki/Languages/eng.resx [hunk 29]
```diff
@@ -2308,4 +2308,23 @@ To read more about the miki scripting language. Check out https://github.com/mik
...
+    <value>You already claimed your daily today! Please wait another `{0}` before using it again.</value>
+  </data>
+  <data name="miki_module_accounts_daily_streak_content" xml:space="preserve">
+    <value>You're on a {0} day daily streak!</value>
+  </data>
+  <data name="miki_module_accounts_daily_streak_title" xml:space="preserve">
+    <value>Streak!</value>
+  </data>
+  <data name="miki_module_accounts_daily_title" xml:space="preserve">
+    <value>💰 Daily</value>
```
Comment (velddev) [inline src/Miki/Languages/eng.resx]: Avoid using emoji in language files

File: src/Miki/Languages/eng.resx [hunk 30]
```diff
@@ -2308,4 +2308,23 @@ To read more about the miki scripting language. Check out https://github.com/mik
...
   </data>
+  <data name="miki_module_accounts_daily_appreciate_list" xml:space="preserve">
+    <value>Vote for us every day on [DiscordBots](https://discordbots.org/bot/160105994217586689/vote) to get an additional bonus!;Donate to us on [Patreon](https://patreon.com/mikibot) for more mekos!</value>
+    <comment>When adding things to this list seperate them with a semicolon (" ; "). </comment>
+  </data>
+  <data name="miki_module_accounts_daily_appreciate_title" xml:space="preserve">
+    <value>Appreciate Miki?</value>
+  </data>
+  <data name="miki_module_accounts_daily_claimed" xml:space="preserve">
+    <value>You already claimed your daily today! Please wait another `{0}` before using it again.</value>
```
Comment (velddev) [inline src/Miki/Languages/eng.resx]: Avoid using markdown styling in language files

File: src/Miki/Services/DailyService.cs [hunk 31]
```diff
@@ -0,0 +1,159 @@
+﻿using Miki.Cache;
+using Miki.Core.Migrations;
+using Miki.Logging;
```
Comment (velddev) [inline src/Miki/Services/DailyService.cs]: Move these inside the namespace

File: src/Miki/Services/DailyService.cs [hunk 32]
```diff
@@ -0,0 +1,159 @@
...
+        /// <inheritdoc />
+        public ValueTask SaveAsync()
+            => unitOfWork.CommitAsync();
+
+        /// <inheritdoc />
+        public void Dispose()
+            => unitOfWork.Dispose();
+    }
+
+    public class DailyClaimResponse
```
Comment (velddev) [inline src/Miki/Services/DailyService.cs]: Move public classes in its own file

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Miki/Services/DailyService.cs  7
- test/Miki.Tests/Services/DailyServiceTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Miki/Services/DailyService.cs  7
- test/Miki.Tests/Services/DailyServiceTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Review by velddev — APPROVED** [review]

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- test/Miki.Tests/Services/DailyServiceTests.cs  1
- src/Miki/Services/DailyService.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Miki/Services/Daily/DailyService.cs  7
- src/Miki/Services/Daily/DailyClaimResponse.cs  1
- test/Miki.Tests/Services/DailyServiceTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity increasing per file
==============================
- src/Miki/Services/Daily/DailyClaimResponse.cs  1
- test/Miki.Tests/Services/DailyServiceTests.cs  1
- src/Miki/Services/Daily/DailyService.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity increasing per file
==============================
- test/Miki.Tests/Services/DailyServiceTests.cs  1
- src/Miki/Modules/FunModule.cs  1
- src/Miki/Services/Daily/DailyClaimResponse.cs  1
- src/Miki/Services/Daily/DailyService.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Miki/Services/Daily/DailyClaimResponse.cs  1
- src/Miki/Services/Daily/DailyService.cs  7
- test/Miki.Tests/Services/DailyServiceTests.cs  1
- src/Miki/Modules/FunModule.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- src/Miki/Services/Daily/DailyClaimResponse.cs  1
- src/Miki/Services/Daily/DailyService.cs  7
- src/Miki/Modules/FunModule.cs  1
- test/Miki.Tests/Services/DailyServiceTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=4603218&bid=15447345)
